lib/streamlit/elements/multiselect.py

```diff
@@ -250,6 +289,13 @@ def _multiselect(
         multiselect_proto.form_id = current_form_id(self.dg)
         if help is not None:
             multiselect_proto.help = dedent(help)
+        if max_selections is not None:
+            if get_default_count(default_value) > max_selections:
+                raise StreamlitAPIException(
+                    getOptionsMessage(get_default_count(default_value), max_selections)
+                )
+            else:
+                multiselect_proto.max_selections = max_selections
```

Comment:
Doesn't this bring us back to square 1 where we're duplicating code here and again after the call to `register_widget`?

Given that we'll catch this below anyway, can't we just always just set

```python
multiselect_proto.max_selections = max_selections
```

before the `if help is not None` statement and rely on the code below to error out when we need it to?

Reply:
fixed.